### PR TITLE
[AL-2918] Add support for FAQTemplate and remove GenericList

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**

### CODE REVIEW

File: Sources/Utilities/ALKMessageViewModel+Extension.swift [hunk 1]
```diff
@@ -9,25 +9,53 @@ import Foundation
...
 
-    func imageMessage() -> ImageMessage? {
-        let payload = self.payloadFromMetadata()
-        precondition(payload != nil, "Payload cannot be nil")
-        guard let imageData = payload?[0], let url = imageData["url"] as? String else {
-            assertionFailure("Payload must contain url.")
-            return nil
-        }
-        let message = Message(
+    private func richMessage() -> Message {
```
Comment (mukeshthawani) [inline Sources/Utilities/ALKMessageViewModel+Extension.swift]: Name can be changed to `messageDetails` or something, as this is not a rich message. And we can make it a computed `var`.

File: RichMessageKit/Views/core/FAQMessageView.swift [hunk 2]
```diff
@@ -0,0 +1,143 @@
+//
+//  FAQMessageView.swift
+//  ApplozicSwift
+//
+//  Created by Shivam Pokhriyal on 03/06/19.
+//
+
+import UIKit
+
```
Comment (mukeshthawani) [inline RichMessageKit/Views/core/FAQMessageView.swift]: Can you document the details of this view like what all elements it contains and how to use it? Like we've done in the case of `SentMessageView` and others.

File: RichMessageKit/Views/SentFAQMessageCell.swift [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        faqStyle: FAQMessageTheme.sentMessage,
+        alignLeft: false)
+
+    fileprivate var messageViewPadding: Padding
+
+    fileprivate lazy var messageViewHeight = messageView.heightAnchor.constraint(equalToConstant: 0)
+
+    static let faqWidth = Config.maxWidth - Config.faqLeftPadding - Config.padding.right
+
+    public override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
```
Comment (mukeshthawani) [inline RichMessageKit/Views/SentFAQMessageCell.swift]: As this is part of RichMessageKit, all the public components can be used separately so, the public properties and initializers should have some documentation.

File: RichMessageKit/Models/FAQMessage.swift [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    public var message: Message
+
+    public var title: String?
+
+    public var description: String?
+
+    public var buttonLabel: String?
+
+    public var buttons: [String]
```
Comment (mukeshthawani) [inline RichMessageKit/Models/FAQMessage.swift]: We can remove the line gap.

File: Sources/Utilities/ALMessage+Extension.swift [hunk 5]
```diff
@@ -95,8 +95,8 @@ extension ALMessage: ALKChatViewModelProtocol {
             return "Text"
         case .genericCard:
             return message
-        case .genericList:
-            return message
+        case .faqTemplate:
+            return message ?? "FAQ"
```
[the remark this answers is not in the document]
Comment (ShivamPokhriyal) [reply Sources/Utilities/ALMessage+Extension.swift]: Will create a separate PR for this as other messages need this change as well.

**Pull request merged**